### PR TITLE
cxx-qt-lib: add option to output C++ to CXX_QT_LIB_OUT_DIR env var

### PR DESCRIPTION
**Pull request by Be-ing**
These C++ files won't build yet because the #include paths need to
be adjusted.

### CODE REVIEW

**Comment by Be-ing** [comment]
I'm thinking the include paths should be:
```cpp
#include <cxx-qt/qttypes.h>
```

**Comment by ahayzen-kdab** [comment]
> I'm thinking the include paths should be:
> 
> ```c++
> #include <cxx-qt/qttypes.h>
> ```

Right, wonder if it should be `cxx-qt` or `cxx-qt-lib` ? As we might have `gen` files too ? eg `#include <cxx-qt-lib/qttypes.h>` and `#include <cxx-qt-gen/my_object.rs.cxxqt.h>` ?

File: cxx-qt-lib/build.rs [hunk 1]
```diff
@@ -80,5 +81,62 @@ fn main() {
...
+            .write_all(cxx_gen::HEADER.as_bytes())
+            .expect("Could not write cxx.h file");
+
+        for class in generated {
+            let include_file_path = PathBuf::from(format!(
+                "{}/{}.h",
+                include_directory_path.display(),
+                class.name
+            ));
+            File::create(&include_file_path)
```
Comment (ahayzen-kdab) [inline cxx-qt-lib/build.rs:126]: We have this file create, write_all, some form of failure block 4 times, DRY points to this being put in a method ?

Or also maybe the whole block needs to be split up, as the header and source paths are repeated for creating the directory, copying the qt_types, creating the individual files.  Maybe there should be one method that does each of those steps once but is passed as parameter to tell it the destination and source ?

**Review by ahayzen-kdab — APPROVED** [review]
LGTM :-)

**Pull request merged**